src/main/kotlin/com/pubnub/api/retry/RetryableBase.kt

```diff
@@ -76,30 +75,28 @@ internal abstract class RetryableBase<T>(
         }
     }
 
-    internal fun getRetryCount(): Int {
-        return when (retryPolicy) {
-            is RequestRetryPolicy.None -> 0
-            is RequestRetryPolicy.Linear -> retryPolicy.maxRetryNumber
-            is RequestRetryPolicy.Exponential -> retryPolicy.maxRetryNumber
-        }
+    protected val maxRetryNumberFromConfiguration: Int = when (retryPolicy) {
+        is RequestRetryPolicy.None -> 0
+        is RequestRetryPolicy.Linear -> retryPolicy.maxRetryNumber
+        is RequestRetryPolicy.Exponential -> retryPolicy.maxRetryNumber
     }
 
     internal fun isErrorCodeRetryable(errorCode: Int) = retryableStatusCodes.containsKey(errorCode)
 
     private fun calculateDelayForTooManyRequestError(
         response: Response<T>
-    ): Int {
+    ): Duration {
         val retryAfterInSec: String? = response.headers()[RETRY_AFTER_HEADER_NAME]
-        if (!retryAfterInSec.isNullOrBlank() && !isNumeric(retryAfterInSec)) {
-            throw PubNubException(PubNubError.RETRY_AFTER_HEADER_VALUE_CAN_NOT_BE_PARSED_TO_INT)
-        }
-        return retryAfterInSec?.toIntOrNull()?.let { retryAfterInSec ->
-            if (retryAfterInSec > 0) {
-                retryAfterInSec * MILLISECONDS
-            } else {
-                getDelayFromRetryPolicy() * MILLISECONDS
+        val delayInSeconds = retryAfterInSec?.toIntOrNull()
+
+        return when {
+            delayInSeconds != null && delayInSeconds > 0 -> {
+                Duration.ofSeconds(delayInSeconds.toLong())
```

Comment:
In my comment I mentioned using `kotlin.time.Duration` and not `java.time.Duration`, so you could do stuff like 
`delay.seconds` and not `Duration.of(...)`
and also use regular operators like + - / *

https://kotlinlang.org/docs/time-measurement.html#create-duration

Reply:
Ok. I will change it.